docs/services.rst

```diff
@@ -424,6 +515,80 @@ it against an existing :term:`Resource`. The resolution of :term:`Effective Perm
+    endpoint representing a `GeoServer`_ instance.
+
+The implementation of this :term:`Service` is handled by class :class:`magpie.services.ServiceGeoserverWPS`.
+It offers similar operations and request handling to :ref:`ServiceWPS`, but adds scoped :class:`magpie.models.Workspace`
+definition of :class:`magpie.models.Process` as for other `GeoServer`_ base :term:`Services <Service>`, in order to
+properly parse request paths that include such references. Other than the :term:`Resource` hierarchy being nested by
+:class:`magpie.models.Workspace`, the rest of the parsing and handling methodology is equivalent to :ref:`ServiceWPS`.
+
+
+ServiceGeoserver
+~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.21
+
+.. seealso::
+    - :ref:`ServiceGeoserverWFS`
+    - :ref:`ServiceGeoserverWMS`
+    - :ref:`ServiceGeoserverWPS`
+
+This :term:`Service` is combined :term:`OWS` implementation for `GeoServer`_ that allows simultaneous representation of
+:term:`WFS`, :term:`WMS` and :term:`WPS` :term:`Resources <Resource>` all nested under a single reference hosted under
+common remote URL. Using this implementation, :class:`magpie.models.Workspace` are first required as immediate children
+under the root :term:`Service`, and can be followed by both :term:`Resources <Resource>` of type
+:class:`magpie.models.Layer` and :class:`magpie.models.Process`.
+There are two main advantages of using this combined implementation over their specific :term:`OWS` counterparts.
+
+First, using the same :term:`Resources <Resource>` to represent corresponding elements in the `GeoServer`_ across
+:term:`OWS` endpoints reduces the chances of inconsistent access to otherwise equivalent :term:`Resources <Resource>`.
+For example, a :class:`magpie.models.Layer` that is granted :term:`Permission` to retrieve features (:term:`WFS`) or
+to render their map raster (:term:`WMS`) will be managed using the same reference. Using distinct
+:ref:`ServiceGeoserverWFS` and :ref:`ServiceGeoserverWMS` :term:`Service` in `Magpie` would require from to
+administrator to always maintain their :term:`Permissions <Permission>` in sync. This is prone to many errors and
+confusion when managing multiple large hierarchies of layers.
+
+Second, this :ref:`ServiceGeoserver` implementation is *configurable*. In other words, if the :term:`Service`
+administrator intends to only make use (for the moment) of :term:`WFS` functionality, they can customize this
+:term:`Service` using the following definition.
+
+.. code-block:: YAML
+
+    providers:
+      RemoteGeoServer:
+        # minimal configuration requirements (where the real `GeoSever` service resides)
+        # other optional parameters from `providers.cfg` can also be provided
+        url: http://localhost:1234
+        type: geoserver
+
+        # customizable configuration (enable desired OWS request handlers)
+        # all OWS are enabled by default if no configuration is provided
+        configuration:
```

Comment:
Is it new that service type can have custom config? Does the UI allow to edit that?

Reply:
No(1) and yes(2). 
Before, only `ServiceTHREDDS` had a customizable config (what defines meta/data access for `browse`/`read` permissions).
Both POST/PATCH requests allow a JSON in `configuration` field. They are then padded with defaults in case of missing contents.

Reply:
Will need to add the option to edit via UI. 
It is possible to define it during service creation, but not modify it afterward.

![image](https://user-images.githubusercontent.com/19194484/151572223-046ac96e-36a5-407e-a766-40d47534422f.png)

Reply:
Ok, it doesn't have to be in that PR though.

Reply:
Yes, I thought so too : #500